server/filter.go

```diff
@@ -62,77 +62,78 @@ func (f *excludedFilter) FilterTarget(store *storeInfo) bool {
 }
 
 type stateFilter struct {
-	cfg *BalanceConfig
+	opt *scheduleOption
 }
 
-func newStateFilter(cfg *BalanceConfig) *stateFilter {
-	return &stateFilter{cfg: cfg}
+func newStateFilter(opt *scheduleOption) *stateFilter {
+	return &stateFilter{opt: opt}
 }
 
-func (sf *stateFilter) filterBadStore(store *storeInfo) bool {
+func (f *stateFilter) filter(store *storeInfo) bool {
 	if !store.isUp() {
 		return true
 	}
-	if store.downTime() >= sf.cfg.MaxStoreDownDuration.Duration {
-		// The store is considered to be down.
-		return true
-	}
-	return false
+	return store.downTime() > f.opt.GetMaxStoreDownTime()
+}
+
+func (f *stateFilter) FilterSource(store *storeInfo) bool {
+	return f.filter(store)
 }
 
-func (sf *stateFilter) FilterSource(store *storeInfo) bool {
-	return sf.filterBadStore(store)
+func (f *stateFilter) FilterTarget(store *storeInfo) bool {
+	return f.filter(store)
 }
 
-func (sf *stateFilter) FilterTarget(store *storeInfo) bool {
-	return sf.filterBadStore(store)
+type regionCountFilter struct {
+	opt *scheduleOption
 }
 
-type capacityFilter struct {
-	cfg *BalanceConfig
+func newRegionCountFilter(opt *scheduleOption) *regionCountFilter {
+	return &regionCountFilter{opt: opt}
 }
 
-func newCapacityFilter(cfg *BalanceConfig) *capacityFilter {
-	return &capacityFilter{cfg: cfg}
+func (f *regionCountFilter) FilterSource(store *storeInfo) bool {
+	return uint64(store.stats.RegionCount) < f.opt.GetMinRegionCount()
 }
 
-func (cf *capacityFilter) FilterSource(store *storeInfo) bool {
-	return store.storageRatio() <= cf.cfg.MinCapacityUsedRatio
+func (f *regionCountFilter) FilterTarget(store *storeInfo) bool {
+	return false
```

Comment:
This filter is used to filter source stores with regions fewer than `MinRegionCount`, it doesn't filter target stores, so it just returns false.